python/lsst/ctrl/oods/fileIngester.py

```diff
@@ -97,21 +96,27 @@ def extract_cause(self, e):
         else:
             return f"{str(e.__cause__)};  {cause}"
 
-    def create_bad_dirname(self, directory, original):
+    def create_bad_dirname(self, bad_dir_root, staging_dir_root, original):
         # strip the original directory location, except for the date
-        newfile = original.lstrip(self.forwarder_staging_dir)
+        newfile = self.strip_prefix(original, staging_dir_root)
 
-        # split into date and filename
+        # split into subdir and filename
         head, tail = os.path.split(newfile)
 
         # create subdirectory path name for directory with date
-        newdir = os.path.join(directory, head)
+        newdir = os.path.join(bad_dir_root, head)
 
         # create the directory, and hand the name back
         os.makedirs(newdir, exist_ok=True)
 
         return newdir
 
+    def strip_prefix(self, name, prefix):
```

Comment:
In ButlerURI I use relative_to:  https://github.com/lsst/daf_butler/blob/master/python/lsst/daf/butler/core/_butlerUri.py#L310-L311

Reply:
I changed the guts of strip_prefix to use this method as well.